### PR TITLE
Restructure the left navigation pane of setup

### PR DESCRIPTION
**Pull request by Rajakavitha1**
Resolves : https://github.com/kubernetes/website/issues/10468

I have restructured the left-nav, changed titles of a few .md files, and resolved all the references in English content.
The intent is to eventually make the entire setup vendor agnostic and the first baby steps are to get the prerequisites (version skew info) then the setup steps in a learning environment and production environment ( vendor agnostic docs under tools )
The current navigation structure does not bring out the work flow to get started with Kubernetes. Yes, there needs to be a lot of work done about the content itself. This PR aims in getting the outline of getting started steps right in the left navigation pane. For example, the user does not have to wade through all the information in the current linear layout before reading about installing minikube if he is a beginner and wants to just setup a Kubernetes Cluster.

Moved kubeadm specific Version skew policy {#version-skew-policy} information in independent/create-cluster-kubeadm.md to https://deploy-preview-14826--kubernetes-io-master-staging.netlify.com/docs/setup/release/version-skew-policy/#version-skew-policy

I updated all the changes in static redirects.
Incorporated the changes suggested in #14558

PS: all the changes and refactoring that has to be done to the existing pages will be addressed in the followup PR. Also the docs URI references can be updated easily after refactoring using the find option in any editor. Add the static redirects also handle the change in URI.

### CODE REVIEW

**Comment by Rajakavitha1** [comment]
> could you please outline all folder moves in a list?
 Here's the list and this has been updated in the redirects page as well.
/docs/setup/release/notes/   no change
/docs/setup/release/notes/

/docs/setup/version-skew-policy/
/docs/setup/release/version-skew-policy/  updated the references

/docs/setup/minikube/
/docs/setup/learning-environment/minikube/ title changed, updated references

/docs/setup/cri/
/docs/setup/production-environment/container-runtimes/ title changed, updated references,

/docs/setup/independent/install-kubeadm/
/docs/setup/production-environment/tools/kubeadm/install-kubeadm/ updated references

/docs/setup/independent/troubleshooting-kubeadm/
/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm/ (updated references)

/docs/setup/independent/create-cluster-kubeadm/
/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/ (updated references)

/docs/setup/independent/control-plane-flags/
/docs/setup/production-environment/tools/kubeadm/control-plane-flags/ (updated references)

/docs/setup/independent/ha-topology/
/docs/setup/production-environment/tools/kubeadm/ha-topology/ updated references

/docs/setup/independent/high-availability/
/docs/setup/production-environment/tools/kubeadm/high-availability/ updated references

/docs/setup/independent/setup-ha-etcd-with-kubeadm/
/docs/setup/production-environment/tools/kubeadm/setup-ha-etcd-with-kubeadm/ (updated references)

/docs/setup/independent/kubelet-integration/
/docs/setup/production-environment/tools/kubeadm/kubelet-integration/ no references

/docs/setup/custom-cloud/kops/
/docs/setup/production-environment/tools/kops/ ( title changed, redirects has one reference in _static redirects)

/docs/setup/custom-cloud/kubespray/
/docs/setup/production-environment/tools/kubespray/ ( title changed, redirects has one reference)

/docs/setup/on-premises-metal/krib/
/docs/setup/production-environment/tools/krib/ (title changed, no references in english)

/docs/setup/turnkey/aws/
/docs/setup/production-environment/turnkey/aws/ (updated references, redirects has one reference)

/docs/setup/turnkey/alibaba-cloud/
/docs/setup/production-environment/turnkey/alibaba-cloud/ (redirects has one reference)

/docs/setup/turnkey/azure/
/docs/setup/production-environment/turnkey/azure/ redirects has one reference

/docs/setup/turnkey/clc/
/docs/setup/production-environment/turnkey/clc/ redirects has one reference

/docs/setup/turnkey/gce/
/docs/setup/production-environment/turnkey/gce/ (updated reference, redirects has one reference)

/docs/setup/turnkey/icp/
/docs/setup/production-environment/turnkey/icp/ no references

/docs/setup/turnkey/stackpoint/
/docs/setup/production-environment/turnkey/stackpoint/ redirects has one reference

/docs/setup/on-premises-vm/cloudstack/
/docs/setup/production-environment/on-premises-vm/cloudstack/ redirects has one reference

/docs/setup/on-premises-vm/dcos/
/docs/setup/production-environment/on-premises-vm/dcos/ redirects has one reference

/docs/setup/on-premises-vm/ovirt/
/docs/setup/production-environment/on-premises-vm/ovirt/ redirects has one reference

/docs/setup/windows/intro-windows-in-kubernetes/
/docs/setup/production-environment/windows/intro-windows-in-kubernetes/ (only in blog)

/docs/setup/windows/user-guide-windows-nodes/
/docs/setup/production-environment/windows/user-guide-windows-nodes/ no references

/docs/setup/windows/user-guide-windows-containers/
/docs/setup/production-environment/windows/user-guide-windows-containers/ no references

/docs/setup/multiple-zones/
/docs/setup/best-practices/multiple-zones/ updated references

/docs/setup/cluster-large/
/docs/setup/best-practices/cluster-large/ updated references

/docs/setup/node-conformance/
/docs/setup/best-practices/node-conformance/ no references

/docs/setup/certificates/
/docs/setup/best-practices/certificates/

**Comment by Rajakavitha1** [comment]
> i have a concert that we are adding way to many different categories like `production-environment` and `learning` while in the end of the day this is about setup-tools and deployers of kubernetes and it's up to the users to figure out how to use them - e.g. kubeadm can still be used for learning.

The idea was to keep in sync with the outline we have in https://kubernetes.io/docs/setup/.

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 1]
```diff
@@ -283,7 +283,15 @@ systemctl restart containerd
 
 To use the `systemd` cgroup driver, set `plugins.cri.systemd_cgroup = true` in `/etc/containerd/config.toml`.
 When using kubeadm, manually configure the
-[cgroup driver for kubelet](/docs/setup/independent/install-kubeadm/#configure-cgroup-driver-used-by-kubelet-on-master-node)
+<<<<<<< HEAD
```
Comment (sftim) [inline content/en/docs/setup/production-environment/container-runtimes.md]: This looks like merge that got left in by accident.

File: content/en/docs/setup/production-environment/tools/kops.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ---
-title: Installing Kubernetes on AWS with kops
+title: Installing Kubernetes with kops
```
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kops.md]: If renaming, how about renaming to `kops-aws.md`?

File: content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md [hunk 3]
```diff
@@ -17,7 +17,11 @@ and control plane nodes are co-located.
 control plane nodes and etcd members are separated.
 
 Before proceeding, you should carefully consider which approach best meets the needs of your applications
-and environment. [This comparison topic](/docs/setup/independent/ha-topology/) outlines the advantages and disadvantages of each.
+<<<<<<< HEAD:content/en/docs/setup/production-environment/tools/independent/high-availability.md
```
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md]: This looks like a merge left in by mistake.

File: content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md [hunk 4]
```diff
@@ -39,10 +43,17 @@ LoadBalancer, or with dynamic PersistentVolumes.
 For both methods you need this infrastructure:
 
 - Three machines that meet [kubeadm's minimum
-  requirements](/docs/setup/independent/install-kubeadm/#before-you-begin) for
+<<<<<<< HEAD:content/en/docs/setup/production-environment/tools/independent/high-availability.md
```
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md]: This looks like a merge left in by mistake.

File: content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md [hunk 5]
```diff
@@ -119,7 +130,11 @@ option. Your cluster requirements may need a different configuration.
 {{< note >}}
 Some CNI network plugins like Calico require a CIDR such as `192.168.0.0/16` and
 some like Weave do not. See the [CNI network
-documentation](/docs/setup/independent/create-cluster-kubeadm/#pod-network).
+<<<<<<< HEAD:content/en/docs/setup/production-environment/tools/independent/high-availability.md
```
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md]: This looks like a merge left in by mistake.

File: content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md [hunk 6]
```diff
@@ -168,7 +183,11 @@ As stated in the command output, the certificate-key gives access to cluster sen
 
 1.  Apply the CNI plugin of your choice:
 
-    [Follow these instructions](/docs/setup/independent/create-cluster-kubeadm/#pod-network) to install
+<<<<<<< HEAD:content/en/docs/setup/production-environment/tools/independent/high-availability.md
```
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md]: This looks like a merge left in by mistake.

File: content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md [hunk 7]
```diff
@@ -211,7 +230,11 @@ in the kubeadm config file.
 
 ### Set up the etcd cluster
 
-1.  Follow [these instructions](/docs/setup/independent/setup-ha-etcd-with-kubeadm/)
+<<<<<<< HEAD:content/en/docs/setup/production-environment/tools/independent/high-availability.md
```
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md]: This looks like a merge left in by mistake.

File: content/en/docs/setup/production-environment/tools/kubeadm/setup-ha-etcd-with-kubeadm.md [hunk 8]
```diff
@@ -26,7 +26,11 @@ when using kubeadm to set up a kubernetes cluster.
 * Some infrastructure to copy files between hosts. For example `ssh` and `scp`
   can satisfy this requirement.
 
-[toolbox]: /docs/setup/independent/install-kubeadm/
+<<<<<<< HEAD:content/en/docs/setup/production-environment/tools/independent/setup-ha-etcd-with-kubeadm.md
```
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/setup-ha-etcd-with-kubeadm.md]: This looks like a merge left in by mistake.

File: content/en/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm.md [hunk 9]
```diff
@@ -58,9 +58,19 @@ This may be caused by a number of problems. The most common are:
   There are two common ways to fix the cgroup driver problem:
 
  1. Install Docker again following instructions
-  [here](/docs/setup/cri/#docker).
+<<<<<<< HEAD:content/en/docs/setup/production-environment/tools/independent/troubleshooting-kubeadm.md
```
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm.md]: This looks like a merge left in by mistake.

File: content/en/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm.md [hunk 10]
```diff
@@ -219,7 +229,15 @@ Error from server: Get https://10.19.0.41:10250/containerLogs/default/mysql-ddc6
 If you have nodes that are running SELinux with an older version of Docker you might experience a scenario
 where the `coredns` pods are not starting. To solve that you can try one of the following options:
 
-- Upgrade to a [newer version of Docker](/docs/setup/cri/#docker).
+<<<<<<< HEAD:content/en/docs/setup/production-environment/tools/independent/troubleshooting-kubeadm.md
```
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm.md]: This looks like a merge left in by mistake.

File: content/en/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm.md [hunk 11]
```diff
@@ -278,7 +296,11 @@ Alternatively, you can try separating the `key=value` pairs like so:
 but this will result in the key `enable-admission-plugins` only having the value of `NamespaceExists`.
 
 A known workaround is to use the kubeadm
-[configuration file](/docs/setup/independent/control-plane-flags/#apiserver-flags).
+<<<<<<< HEAD:content/en/docs/setup/production-environment/tools/independent/troubleshooting-kubeadm.md
```
Comment (sftim) [inline content/en/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm.md]: This looks like a merge left in by mistake.

File: content/en/docs/setup/production-environment/windows/user-guide-windows-nodes.md [hunk 12]
```diff
@@ -21,7 +21,11 @@ The Kubernetes platform can now be used to run both Linux and Windows containers
 ## Before you begin
 
 * Obtain a [Windows Server license](https://www.microsoft.com/en-us/cloud-platform/windows-server-pricing) in order to configure the Windows node that hosts Windows containers. You can use your organization's licenses for the cluster, or acquire one from Microsoft, a reseller, or via the major cloud providers such as GCP, AWS, and Azure by provisioning a virtual machine running Windows Server through their marketplaces. A [time-limited trial](https://www.microsoft.com/en-us/cloud-platform/windows-server-trial) is also available.
-* Build a Linux-based Kubernetes cluster in which you have access to the control plane (some examples include [Getting Started from Scratch](/docs/setup/release/building-from-source/), [kubeadm](/docs/setup/independent/create-cluster-kubeadm/), [AKS Engine](/docs/setup/turnkey/azure/), [GCE](/docs/setup/turnkey/gce/), [AWS](/docs/setup/turnkey/aws/)).
+<<<<<<< HEAD:content/en/docs/setup/production-environment/windows/user-guide-windows-nodes.md
```
Comment (sftim) [inline content/en/docs/setup/production-environment/windows/user-guide-windows-nodes.md]: This looks like a merge left in by mistake.

**Comment by neolit123** [comment]
> Here's the list and this has been updated in the redirects page as well.

thanks.
given the tools for `/docs/setup/production-environment` are under `tools/`
should `/docs/setup/learning-environment/minikube/` be `/docs/setup/learning-environment/tools/minikube/` ?

File: content/en/docs/setup/_index.md [hunk 13]
```diff
@@ -40,7 +40,7 @@ If you're learning Kubernetes, use the Docker-based solutions: tools supported b
 
 |Community           |Ecosystem     |
 | ------------       | --------     |
-| [Minikube](/docs/setup/minikube/) | [CDK on LXD](https://www.ubuntu.com/kubernetes/docs/install-local) |
+| [Minikube](/docs/setup/learning-environment/minikube/) | [CDK on LXD](https://www.ubuntu.com/kubernetes/docs/install-local) |
```
[the remark this answers is not in the document]
Comment (afbjorklund) [reply content/en/docs/setup/_index.md]: I'm thinking "yes", at least until minikube starts deploying multi-node clusters...

> The primary goal of minikube is to make it simple to run Kubernetes locally, for day-to-day development workflows and l**earning purpose**

^ Emphasis mine

And it is already sorted under "Learning Environment" header, this is just URL

File: content/en/docs/setup/production-environment/tools/kubeadm/control-plane-flags.md [hunk 14]
```diff
@@ -23,7 +23,7 @@ The `extraArgs` field consist of `key: value` pairs. To override a flag for a co
 2.  Add the flags to override to the field.
 
 For more details on each field in the configuration you can navigate to our
-[API reference pages](https://godoc.org/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta1).
+[API reference pages](https://godoc.org/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm#ClusterConfiguration).
```
[the remark this answers is not in the document]
Comment (Rajakavitha1) [reply content/en/docs/setup/production-environment/tools/kubeadm/control-plane-flags.md]: updated as suggested.

File: content/en/docs/setup/production-environment/container-runtimes.md (hunk 1 above)
Comment (Rajakavitha1) [reply content/en/docs/setup/production-environment/container-runtimes.md]: Thank you it was a miss. Fixed it now.

File: content/en/docs/setup/production-environment/tools/kops.md (hunk 2 above)
Comment (Rajakavitha1) [reply content/en/docs/setup/production-environment/tools/kops.md]: Not sure if we want any vendor specific docs in future at all. I will leave this as is for now.

File: content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md (hunk 3 above)
Comment (Rajakavitha1) [reply content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md]: Thank you it was a miss. Fixed it now.

File: content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md (hunk 4 above)
Comment (Rajakavitha1) [reply content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md]: Thank you it was a miss. Fixed it now.

File: content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md (hunk 5 above)
Comment (Rajakavitha1) [reply content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md]: Thank you it was a miss. Fixed it now.

File: content/en/docs/setup/production-environment/tools/kubeadm/setup-ha-etcd-with-kubeadm.md (hunk 8 above)
Comment (Rajakavitha1) [reply content/en/docs/setup/production-environment/tools/kubeadm/setup-ha-etcd-with-kubeadm.md]: Thank you it was a miss. Fixed it now.

File: content/en/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm.md (hunk 9 above)
Comment (Rajakavitha1) [reply content/en/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm.md]: Thank you it was a miss. Fixed it now.

File: content/en/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm.md (hunk 10 above)
Comment (Rajakavitha1) [reply content/en/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm.md]: Thank you it was a miss. Fixed it now.

File: content/en/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm.md (hunk 11 above)
Comment (Rajakavitha1) [reply content/en/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm.md]: Thank you it was a miss. Fixed it now.

File: content/en/docs/setup/production-environment/windows/user-guide-windows-nodes.md (hunk 12 above)
Comment (Rajakavitha1) [reply content/en/docs/setup/production-environment/windows/user-guide-windows-nodes.md]: Thank you it was a miss. Fixed it now.

File: content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md (hunk 6 above)
Comment (Rajakavitha1) [reply content/en/docs/setup/production-environment/tools/kubeadm/high-availability.md]: Thank you it was a miss. Fixed it now.

> > Here's the list and this has been updated in the redirects page as well.
> 
> thanks.
> given the tools for `/docs/setup/production-environment` are under `tools/`
> should `/docs/setup/learning-environment/minikube/` be `/docs/setup/learning-environment/tools/minikube/` ?

Hmm... good point. for now there is only minikube in learning-environment. As we get more content for kind and others we could add a new tools directory.

**Comment by Rajakavitha1** [comment]
Hi @sftim and @neolit123  incorporated all the changes suggested.
Please take a look and state your approval. 

PS: This PR involves lot of files that are constantly updated. It is little daunting task to always rebase and resolve conflicts.

**Comment by tengqm** [comment]
This one is huge ... unnecessarily huge. We could break it into a few smaller PRs for 1) easier review and 2) less conflicts. However, I vote for merging this and leaving further improvement to future PRs.
/lgtm

**Comment by sftim** [comment]
/lgtm
as well

**Comment by Rajakavitha1** [comment]
> This one is huge ... unnecessarily huge. We could break it into a few smaller PRs for 1) easier review and 2) less conflicts. However, I vote for merging this and leaving further improvement to future PRs.
> /lgtm

Thanks @tengqm !!!

**Comment by jimangel** [comment]
/approve

Let's get this in and work on top of it, as said above. Thanks!

**Comment by DanyC97** [comment]
great work @Rajakavitha1 !

**Pull request merged**